test/limitOrderBook/limitOrderBook.fillOrder.test.ts

```diff
@@ -0,0 +1,575 @@
+        limitOrderFeeVault = fixture.limitOrderFeeVault
+        rewardToken = fixture.rewardToken
+
+        const pool1LowerTick: number = priceToTick(2000, await pool.tickSpacing())
+        const pool1UpperTick: number = priceToTick(4000, await pool.tickSpacing())
+        delegateApproval = fixture.delegateApproval
+        limitOrderBook = fixture.limitOrderBook
+
+        // ETH
+        await initAndAddPool(
+            fixture,
+            pool,
+            baseToken.address,
+            encodePriceSqrt("2960", "1"),
+            10000, // 1%
+            // set maxTickCrossed as maximum tick range of pool by default, that means there is no over price when swap
+            getMaxTickRange(),
+        )
+        await syncIndexToMarketPrice(mockedBaseAggregator, pool)
+
+        // prepare collateral for maker
+        await mintAndDeposit(fixture, maker, 1_000_000_000_000)
+
+        // maker add liquidity
+        await clearingHouse.connect(maker).addLiquidity({
+            baseToken: baseToken.address,
+            base: parseEther("3000"),
+            quote: parseEther("10000000"),
+            lowerTick: pool1LowerTick,
+            upperTick: pool1UpperTick,
+            minBase: 0,
+            minQuote: 0,
+            useTakerBalance: false,
+            deadline: ethers.constants.MaxUint256,
+        })
+
+        // prepare collateral for taker
+        await mintAndDeposit(fixture, trader, 1000)
+
+        await delegateApproval.connect(trader).approve([
+            {
+                delegate: limitOrderBook.address,
+                action: fixture.clearingHouseOpenPositionAction,
+            },
+        ])
+    })
+
+    it("fill limit order successfully", async () => {
+        // long 0.1 ETH at $3000 with $300
+        const limitOrder = {
+            orderType: fixture.orderTypeLimitOrder,
+            salt: 1,
+            trader: trader.address,
+            baseToken: baseToken.address,
+            isBaseToQuote: false,
+            isExactInput: true,
+            amount: parseEther("300").toString(),
+            oppositeAmountBound: parseEther("0.1").toString(),
+            deadline: ethers.constants.MaxUint256.toString(),
+            reduceOnly: false,
+            roundIdWhenCreated: parseEther("0").toString(),
+            triggerPrice: parseEther("0").toString(),
+        }
+
+        // sign limit order
+        const signature = await getSignature(fixture, limitOrder, trader)
+        const orderHash = await getOrderHash(fixture, limitOrder)
+
+        const tx = await limitOrderBook.connect(keeper).fillLimitOrder(limitOrder, signature, parseEther("0"))
+        await expect(tx)
+            .to.emit(limitOrderBook, "LimitOrderFilled")
+            .withArgs(trader.address, baseToken.address, orderHash, keeper.address, fixture.rewardAmount)
+
+        await expect(tx).to.emit(limitOrderFeeVault, "Disbursed").withArgs(keeper.address, fixture.rewardAmount)
```

Comment:
can add one more check, check keeper's balance to make sure the keeper really get reward.